core/src/main/scala/kafka/controller/ControllerContext.scala

```diff
@@ -328,8 +328,9 @@ class ControllerContext {
   }
 
   def queueTopicDeletion(topics: Set[String]): Unit = {
+    val newlyDeletedTopics = topics.diff(topicsToBeDeleted)
     topicsToBeDeleted ++= topics
```

Comment:
Since we've calculated the diff, we can directly add the `newlyDeletedTopics` to avoid unnecessary duplication, right?